### PR TITLE
[RFR] Adding “Memoize” entry

### PR DESCRIPTION
**Pull request by kostasx**

### CODE REVIEW

File: glossary/MEMOIZE.md [hunk 1]
```diff
@@ -0,0 +1,36 @@
+# Memoize
+
+An optimization used to speed up consecutive function calls by caching the result of calls with identical input.
+
+Here is an example of a memoizer function, taken from the book **JavaScript The Good Parts** by Douglas Crockford,
+that caches the results from a fibonacci number generator function:
+
+```
```
Comment (KittyGiraudel) [inline glossary/MEMOIZE.md]: Could you add `js` for syntax highlighting? Also, I started with 2 spaces indentation in other code snippets. Would be nice if we could keep it this way for consistency. :)

**Comment by KittyGiraudel** [comment]
Cool, thanks!

**Comment by kostasx** [comment]
Just added JS syntax and 2 spaces indentation.

**Comment by KittyGiraudel** [comment]
Thanks! :)

**Comment by kostasx** [comment]
Thanks to you too for maintaining this cool project.
;)

**Pull request merged**